### PR TITLE
Add headers concept to Counterfact, add example user route

### LINKED ISSUE #21: Headers in the request and response
Counterfact should be able to read HTTP request headers and send HTTP response headers.

### PR DESCRIPTION
**Pull request by KameronKales**
This is a very basic implementation of headers in Counterfact. If we go with this route, you can send headers in any route, and get them returned to you on the response. 

I added an example implementation under `demo/user.js` to decode a JWT sent in an authorization header. 

There wasnt much to actually test here since the Counterfact side is really simple. 

closes #21

### CODE REVIEW

File: test/dispatcher.test.js [hunk 1]
```diff
@@ -48,6 +48,31 @@ describe("a dispatcher", () => {
     expect(response.body).toBe("Hello Catherine of Aragon!");
   });
 
+  it("passes the request headers", async () => {
+    const registry = new Registry();
+    const mockedJWT =
+      // eslint-disable-next-line max-len
+      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c";
```
Comment (KameronKales) [inline test/dispatcher.test.js]: This seems kind of ugly to put everywhere. Id be happy to extract out to a shared constants folder or something along those lines if we want to.

File: test/dispatcher.test.js [hunk 2]
```diff
@@ -48,6 +48,31 @@ describe("a dispatcher", () => {
...
+        };
+      },
+    });
+
+    const dispatcher = new Dispatcher(registry);
+    const response = await dispatcher.request({
+      method: "GET",
+      path: "/a",
+
+      headers: mockedJWT,
```
Comment (pmcelhaney) [inline test/dispatcher.test.js]: `headers` should be an object

```suggestion
      headers: { 
        Authorization: `Bearer: ${mockedJWT}`, 
        "Cache-Control": "max-age=0",
        // etc
      }, 
```

File: test/dispatcher.test.js (hunk 1 above)
Comment (pmcelhaney) [reply test/dispatcher.test.js]: The unit test isn't doing anything with the JWT. You could use any old [request header](https://developer.mozilla.org/en-US/docs/Glossary/Request_header) (or combination of headers) to test.

**Review by pmcelhaney — CHANGES_REQUESTED** [review]
Thanks! Looks good, just need to update the unit test.

File: test/dispatcher.test.js (hunk 2 above)
Comment (pmcelhaney) [reply test/dispatcher.test.js]: Hence #76 😆

**Review by pmcelhaney — APPROVED** [review]

**Pull request merged**